README.md
[![Apache 2](http://img.shields.io/badge/license-Apache%202-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0)

# Plesk Business & Collaboration Edition 

This repository contains all you need to build and customize your personal preconfigured Business & Collaboration Edition with Plesk. The included whitepaper describes the solution in detail and highlights why you should offer it to your customers.

![](https://raw.githubusercontent.com/plesk/ext-welcome-guide/master/src/_meta/screenshots/1.png)

## Description

Here at Plesk we strive to Simpify the Lives of Web Professionals. This includes our Partners. This cookbook will install and configure Plesk Core, Modules and Bundle Extensions. It will also enable and configure security features and prepare your VM to be used as a VM template. 

## Requirements

 * Contract with Plesk to be able to retrieve Plesk Licenses from the Key Administrator Server (KA)
 * Provide Plesk Key with Proper Plesk Exension keys associated in cookbook script

 or

 * A Plesk license with the [Business & Collaboration Pack](https://ext.plesk.com/packages/63d4feb8-b756-4c64-a99d-c9be0e1ce982-offer-business-feature-pack)
 
## Deploying Plesk with Business & Collaboration Edition Cookbook

1. Install one of the supported Linux Operating Systems ( currently RHEL7, CentOS7, and Debian Jessy) with following hardware specs ( https://docs.plesk.com/release-notes/onyx/hardware-requirements/ )

2. Shell to server as root

3. Download and Edit Variables noted in install_business_edition.sh

4. Execute command 

   $ chmod +x install install_business_edition.sh

5. Execute command 

   $ ./install_business_edition.sh <YOUR_ACTIVATION_KEY>

### Using prebuild Plesk Images

tbd

## Support

Found a bug or experienced an issue? Please open an [issue on Github](https://github.com/plesk/business-and-collaboration-edition/issues)!


CHANGES.md
# 1.2.1
* [*] Fixed bug with plesk_ui

# 1.2
* [*] Replaced old Welcome Extension with the new Welcome Guide

# 1.1
* [*] Updated the cookbook for the latest version of the Business & Collaboration Edition
* [*] Replaced the Security Advisor with our new Advisor Extension
* [*] Replaced PHP5.6 with PHP7.2
* [*] Changed default behaviour of cloning to false
* [*] New version of the welcome extension was added 
* [+] Added Extension installation for SEO Toolkit, Advisor, Uptime Robot and many more 
* [+] Added Solution Type for reporting purposes
* [+] Mentioned Feature Pack in README.md 
* [+] Added option to use the activation key as parameter
* [+] Added output of one-time login link if clone is set to off
* [-] Removed extensions no longer available in Business & Collaboration Edition like MagicSpam
* [-] Removed Route 53 Extension
* [-] Removed no longer needed admin information

# 1.0.1

* [*] Updated link to the latest Welcome Business extension package (version 1.0.1)

# 1.0.0

* [+] First commit

install_business_edition.sh
#!/bin/bash

################################################################################
##          Installation script for Plesk Business & Collaboration Edition    ##
################################################################################

# Edit variables for Plesk pre-configuration

email='admin@test.tst'
passwd='CookBook123'
name='admin'
agreement=true


# Plesk Activation Code - provide proper license for initialization, it will be replaced after cloning
# leave as null if not providing key
activation_key=$1

# Plesk UI View - can be set to Service Provider View (spv) or Power User View (puv)
plesk_ui=spv

# Turn on Fail2Ban, yes or no, Keep in mind you need to provide temp license for initialization for this to work
fail2ban=yes

# Turn on http2
http2=yes

# Turn on Cloning - Set to "on" if this it to make a Golden Image, set to "off" if for remote installation
clone=off

# Test to make sure all initialization values are set

if [[ -z $activation_key ]]; then
echo 'Please provide a proper Plesk Activation Code (Bundle License).'
  exit 1
fi

if [[ -z $email || -z $passwd || -z $name || -z $agreement ]]; then
  echo 'One or more variables are undefined. Please check your initialization values.'
  exit 1
fi

echo "Plesk initialization values are all assigned. We are good so far."
echo

######### Do not edit below this line ###################
#########################################################

# Download Plesk AutoInstaller

echo "Downloading Plesk Auto-Installer"
wget https://installer.plesk.com/plesk-installer
echo

# Make Installed Executable

echo "Making Plesk Auto-Installer Executable"
chmod +x ./plesk-installer
echo

# Install Plesk with Required Components

echo "Starting Plesk Installation"
./plesk-installer install plesk --preset Recommended --with fail2ban modsecurity spamassassin mailman psa-firewall pmm health-monitor kav
echo
echo

# Initalize Plesk before Additional Configuration
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/init_conf-server-configuration.37843/

echo "Starting initialization process of your new Plesk Business & Collaboration Edition"
plesk bin init_conf --init -name $name -email $email -passwd $passwd -license_agreed $agreement
plesk bin settings --set solution_type="business"
echo

# Install Plesk Activation Key if provided
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/license-license-keys.71029/

if [[ -n "$activation_key" ]]; then
  echo "Installing Plesk Activation Code"
  plesk bin license --install $activation_key
  echo
fi

# Configure Service Provider View On

if [ "$plesk_ui" = "spv" ]; then
    echo "Setting to Service Provider View"
    plesk bin poweruser --off
    echo
else
    echo "Setting to Power user View"
    plesk bin poweruser --on
    echo
fi

# Make sure Plesk UI and Plesk Update ports are allowed

echo "Setting Firewall to allow proper ports."
iptables -I INPUT -p tcp --dport 21 -j ACCEPT
iptables -I INPUT -p tcp --dport 22 -j ACCEPT
iptables -I INPUT -p tcp --dport 25 -j ACCEPT
iptables -I INPUT -p tcp --dport 80 -j ACCEPT
iptables -I INPUT -p tcp --dport 110 -j ACCEPT
iptables -I INPUT -p tcp --dport 143 -j ACCEPT
iptables -I INPUT -p tcp --dport 443 -j ACCEPT
iptables -I INPUT -p tcp --dport 465 -j ACCEPT
iptables -I INPUT -p tcp --dport 993 -j ACCEPT
iptables -I INPUT -p tcp --dport 995 -j ACCEPT
iptables -I INPUT -p tcp --dport 8443 -j ACCEPT
iptables -I INPUT -p tcp --dport 8447 -j ACCEPT
iptables -I INPUT -p tcp --dport 8880 -j ACCEPT
echo

# Enable Modsecurity
# https://docs.plesk.com/en-US/onyx/administrator-guide/server-administration/web-application-firewall-modsecurity.73383/

echo "Turning on Modsecurity WAF Rules"
plesk bin server_pref --update-web-app-firewall -waf-rule-engine on -waf-rule-set tortix -waf-rule-set-update-period daily -waf-config-preset tradeoff
echo

# Enable Fail2Ban and Jails
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/ip_ban-ip-address-banning-fail2ban.73594/

if [ "$fail2ban" = "yes" ]; then
  echo "Configuring Fail2Ban and its Jails"
  plesk bin ip_ban --enable
  plesk bin ip_ban --enable-jails ssh
  plesk bin ip_ban --enable-jails recidive
  plesk bin ip_ban --enable-jails modsecurity
  plesk bin ip_ban --enable-jails plesk-proftpd
  plesk bin ip_ban --enable-jails plesk-postfix
  plesk bin ip_ban --enable-jails plesk-roundcube
  plesk bin ip_ban --enable-jails plesk-apache-badbot
  plesk bin ip_ban --enable-jails plesk-panel
  plesk bin ip_ban --enable-jails plesk-wordpress
  plesk bin ip_ban --enable-jails plesk-dovecot 
  plesk bin ip_ban --enable-jails plesk-apache
  plesk bin ip_ban --enable-jails plesk-horde
  echo
fi

# Turn on http2
# https://docs.plesk.com/en-US/onyx/administrator-guide/web-servers/apache-and-nginx-web-servers-linux/http2-support-in-plesk.76461/

if [ "$http2" = "yes" ]; then
  echo "Activating http2"
  /usr/sbin/plesk bin http2_pref --enable
  echo
fi

# Install Bundle Extensions
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/extension-extensions.71031/

echo "Installing Requested Plesk Extensions"
echo
echo "Installing SEO Toolkit"
plesk bin extension --install-url https://ext.plesk.com/packages/2ae9cd0b-bc5c-4464-a12d-bd882c651392-xovi/download
echo
echo "Installing BoldGrid"
plesk bin extension --install-url https://ext.plesk.com/packages/e4736f87-ba7e-4601-a403-7c82682ef07d-boldgrid/download
echo
echo "Installing Acronis Backup"
plesk bin extension --install-url https://ext.plesk.com/packages/93b3b2a8-bd56-4371-8ab9-63c604efab56-acronis-backup/download
echo
echo "Installing Revisium Antivirus"
plesk bin extension --install-url https://ext.plesk.com/packages/b71916cf-614e-4b11-9644-a5fe82060aaf-revisium-antivirus/download
echo
echo "Installing Backup to Cloud extensions"
plesk bin extension --install-url https://ext.plesk.com/packages/9f3b75b3-d04d-44fe-a8fa-7e2b1635c2e1-dropbox-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/52fd6315-22a4-48b8-959d-b2f1fd737d11-google-drive-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/8762049b-870e-47cb-ba14-9f055b99b508-s3-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/a8e5ad9c-a254-4bcf-8ae4-5440f13a88ad-one-drive-backup/download
echo
echo "Installing Pagespeed Insights"
plesk bin extension --install-url https://ext.plesk.com/packages/3d2639e6-64a9-43fe-a990-c873b6b3ec66-pagespeed-insights/download
echo
echo "Installing Uptime Robot"
plesk bin extension --install-url https://ext.plesk.com/packages/7d37cfde-f133-4085-91ea-d5399862321b-uptime-robot/download
echo
echo "Installing Plesk Premium Email powered by Kolab"
plesk bin extension --install-url https://ext.plesk.com/packages/bdc94ad6-cc46-4f5a-8ee7-76675fc3cc44-kolab/download
echo
echo "Configuring Plesk Premium Email powered by Kolab. The Plesk Premium Email license must be installed before this command"
/usr/local/psa/admin/bin/php -dauto_prepend_file=sdk.php /usr/local/psa/admin/plib/modules/kolab/scripts/unattendedinstall.php
echo 
echo "Installing Domain Connect"
plesk bin extension --install-url https://ext.plesk.com/packages/3a36f828-e477-4600-be33-48c21e351c9a-domain-connect/download
echo
echo "Installing Welcome Extension"
plesk bin extension --install-url https://ext.plesk.com/packages/39eb8f3d-0d9a-4605-a42a-c37ca5809415-welcome/download
echo

echo "Enabling Welcome Guide for the Plesk Business & Collaboration Edition"
plesk ext welcome --select -preset business
echo

# Prepare for Cloning
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/cloning-server-cloning-settings.71035/

if [ "$clone" = "on" ]; then
	echo "Setting Plesk Cloning feature."
	plesk bin cloning --update -prepare-public-image true
	echo "Plesk initialization will be wiped on next boot. Ready for Cloning."
else
  echo "Here is your login"
  plesk login
fi

echo
echo "Your Plesk Business & Collaboration Edition image is complete."
echo "Thank you for using the Business  & Collaboration Edition Cookbook"
echo


install_business_server.sh
#!/bin/bash

################################################################################
##          Installation script for Plesk Business Server                     ##
################################################################################

# Edit variables for Plesk pre-configuration

hostname='cp.domain.tst'
email='admin@test.tst'
passwd='CookBook123'
name='admin'
agreement=true
ip_type=shared

# Plesk Activation Code - provide proper license for initialization, it will be replaced after cloning
# leave as null if not providing key
activation_key=$1

# Plesk UI View - can be set to Service Provider View (spv) or Power User View (puv)
plesk_ui=spv

# Turn on Fail2Ban, yes or no, Keep in mind you need to provide temp license for initialization for this to work
fail2ban=yes

# Turn on http2
http2=yes

# Turn on Cloning - Set to "on" if this it to make a Golden Image, set to "off" if for remote installation
clone=off

# Test to make sure all initialization values are set

if [[ -z $activation_key ]]; then
echo 'Please provide a proper Plesk Activation Code (Bundle License).'
  exit 1
fi

if [[ -z $hostname || -z $email || -z $passwd || -z $name || -z $agreement || -z $ip_type ]]; then
  echo 'One or more variables are undefined. Please check your initialization values.'
  exit 1
fi

echo "Plesk initialization values are all assigned. We are good so far."
echo

######### Do not edit below this line ###################
#########################################################

# Download Plesk AutoInstaller

echo "Downloading Plesk Auto-Installer"
wget https://installer.plesk.com/plesk-installer
echo

# Make Installed Executable

echo "Making Plesk Auto-Installer Executable"
chmod +x ./plesk-installer
echo

# Install Plesk with Required Components

echo "Starting Plesk Installation"
./plesk-installer install plesk --preset Recommended --with panel bind fail2ban l10n pmm mysqlgroup roundcube kav spamassassin postfix dovecot proftpd awstats modsecurity mod_fcgid webservers php7.2 php7.1 config-troubleshooter psa-firewall heavy-metal-skin letsencrypt 
echo
echo

# Initalize Plesk before Additional Configuration
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/init_conf-server-configuration.37843/

echo "Starting initialization process of your new Plesk Business server"
plesk bin init_conf --init -email $email -passwd $passwd -hostname $hostname -license_agreed $agreement -ip-type $ip_type
plesk bin settings --set solution_type="business"
echo

# Install Plesk Activation Key if provided
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/license-license-keys.71029/

if [[ -n "$activation_key" ]]; then
  echo "Installing Plesk Activation Code"
  plesk bin license --install $activation_key
  echo
fi

# Configure Service Provider View On

if [ "$plesk_ui" = "spv" ]; then
    echo "Setting to Service Provider View"
    plesk bin poweruser --off
    echo
else
    echo "Setting to Power user View"
    plesk bin poweruser --on
    echo
fi

# Make sure Plesk UI and Plesk Update ports are allowed

echo "Setting Firewall to allow proper ports."
iptables -I INPUT -p tcp --dport 21 -j ACCEPT
iptables -I INPUT -p tcp --dport 22 -j ACCEPT
iptables -I INPUT -p tcp --dport 25 -j ACCEPT
iptables -I INPUT -p tcp --dport 80 -j ACCEPT
iptables -I INPUT -p tcp --dport 110 -j ACCEPT
iptables -I INPUT -p tcp --dport 143 -j ACCEPT
iptables -I INPUT -p tcp --dport 443 -j ACCEPT
iptables -I INPUT -p tcp --dport 465 -j ACCEPT
iptables -I INPUT -p tcp --dport 993 -j ACCEPT
iptables -I INPUT -p tcp --dport 995 -j ACCEPT
iptables -I INPUT -p tcp --dport 8443 -j ACCEPT
iptables -I INPUT -p tcp --dport 8447 -j ACCEPT
iptables -I INPUT -p tcp --dport 8880 -j ACCEPT
echo

# Enable Modsecurity
# https://docs.plesk.com/en-US/onyx/administrator-guide/server-administration/web-application-firewall-modsecurity.73383/

echo "Turning on Modsecurity WAF Rules"
plesk bin server_pref --update-web-app-firewall -waf-rule-engine on -waf-rule-set tortix -waf-rule-set-update-period daily -waf-config-preset tradeoff
echo

# Enable Fail2Ban and Jails
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/ip_ban-ip-address-banning-fail2ban.73594/

if [ "$fail2ban" = "yes" ]; then
  echo "Configuring Fail2Ban and its Jails"
  plesk bin ip_ban --enable
  plesk bin ip_ban --enable-jails ssh
  plesk bin ip_ban --enable-jails recidive
  plesk bin ip_ban --enable-jails modsecurity
  plesk bin ip_ban --enable-jails plesk-proftpd
  plesk bin ip_ban --enable-jails plesk-postfix
  plesk bin ip_ban --enable-jails plesk-dovecot
  plesk bin ip_ban --enable-jails plesk-roundcube
  plesk bin ip_ban --enable-jails plesk-roundcube
  plesk bin ip_ban --enable-jails plesk-apache-badbot
  plesk bin ip_ban --enable-jails plesk-panel
  plesk bin ip_ban --enable-jails plesk-wordpress
  echo
fi

# Turn on http2
# https://docs.plesk.com/en-US/onyx/administrator-guide/web-servers/apache-and-nginx-web-servers-linux/http2-support-in-plesk.76461/

if [ "$http2" = "yes" ]; then
  echo "Activating http2"
  /usr/sbin/plesk bin http2_pref --enable
  echo
fi

# Install Bundle Extensions
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/extension-extensions.71031/

echo "Installing Requested Plesk Extensions"
echo "Installing WordPress Toolkit"
plesk bin extension --install-url https://ext.plesk.com/packages/00d002a7-3252-4996-8a08-aa1c89cf29f7-wp-toolkit/download
echo
echo "Installing SEO Toolkit"
plesk bin extension --install-url https://ext.plesk.com/packages/2ae9cd0b-bc5c-4464-a12d-bd882c651392-xovi/download
echo
echo "Installing Advisor"
plesk bin extension --install-url https://ext.plesk.com/packages/bbf16bc7-094e-4cb3-8b9c-32066fc66561-advisor/download
echo
echo "Installing BoldGrid"
plesk bin extension --install-url https://ext.plesk.com/packages/e4736f87-ba7e-4601-a403-7c82682ef07d-boldgrid/download
echo
echo "Installing Acronis Backup"
plesk bin extension --install-url https://ext.plesk.com/packages/93b3b2a8-bd56-4371-8ab9-63c604efab56-acronis-backup/download
echo
echo "Installing Revisium Antivirus"
plesk bin extension --install-url https://ext.plesk.com/packages/b71916cf-614e-4b11-9644-a5fe82060aaf-revisium-antivirus/download
echo
echo "Installing Backup to Cloud Pro"
plesk bin extension --install-url https://ext.plesk.com/packages/9f3b75b3-d04d-44fe-a8fa-7e2b1635c2e1-dropbox-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/52fd6315-22a4-48b8-959d-b2f1fd737d11-google-drive-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/8762049b-870e-47cb-ba14-9f055b99b508-s3-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/a8e5ad9c-a254-4bcf-8ae4-5440f13a88ad-one-drive-backup/download
echo
echo "Installing Pagespeed Insights"
plesk bin extension --install-url https://ext.plesk.com/packages/3d2639e6-64a9-43fe-a990-c873b6b3ec66-pagespeed-insights/download
echo
echo "Installing Uptime Robot"
plesk bin extension --install-url https://ext.plesk.com/packages/7d37cfde-f133-4085-91ea-d5399862321b-uptime-robot/download
echo
echo "Installing Plesk Premium Email powered by Kolab"
plesk bin extension --install-url https://ext.plesk.com/packages/bdc94ad6-cc46-4f5a-8ee7-76675fc3cc44-kolab/download
echo
echo "Installing LetsEncrypt"
plesk bin extension --install-url https://ext.plesk.com/packages/f6847e61-33a7-4104-8dc9-d26a0183a8dd-letsencrypt/download
echo
echo "Installing Migration Manager"
plesk bin extension --install-url https://ext.plesk.com/packages/bebc4866-d171-45fb-91a6-4b139b8c9a1b-panel-migrator/download
echo
echo "Installing Welcome Extension"
plesk bin extension --install-url https://github.com/plesk/ext-welcome-business/releases/download/v1.0.2/ext-welcome-business_v1.0.2-4.zip
echo


# Prepare for Cloning
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/cloning-server-cloning-settings.71035/

if [ "$clone" = "on" ]; then
	echo "Setting Plesk Cloning feature."
	plesk bin cloning --update -prepare-public-image true
	echo "Plesk initialization will be wiped on next boot. Ready for Cloning."
else
  echo "Here is your login"
  plesk login
fi


echo
echo "Your Plesk Business Server image is complete."
echo "Thank you for using the Business Server Cookbook"
echo
echo
